lib/pavilion/test_config/resolver.py

```diff
@@ -1159,3 +1165,21 @@ def resolve_section_values(cls, component, var_man, allow_deferred=False,
             raise TestConfigError("Invalid value type '{}' for '{}' when "
                                   "resolving strings."
                                   .format(type(component), component))
+
+def resolve_cmd_inheritance(test_cfg):
+
+    for section in ['build', 'run']:
+        config = test_cfg.get(section)
+        if not config:
+            continue
+        new_cmd_list = []
+        if config.get('cmds_extend_before'):
+            new_cmd_list += config.get('cmds_extend_before')
+            config['cmds_extend_before'] = []
+        new_cmd_list += test_cfg[section]['cmds']
+        if config.get('cmds_extend_after'):
+            new_cmd_list += config.get('cmds_extend_after')
```

Comment:
This is fine here, but if you're doing a lot of list extensions (especially with very long lists), you should use the `.extend()` method.
Your way has to copy both lists into a new list, which takes the time and memory equal to the size of both.
`.extend()` just adds to an existing list, and only consumes resources based on the size of what you're adding.